src/SDK/Common/Retry/RetryPolicyInterface.php

```diff
@@ -0,0 +1,131 @@
+ *    $this->setRetryPolicy(ExponentialWithJitterRetryPolicy::getDefault());
+ *
+ *    User can provide different values for retry params as well as follows:
+ *    $this->setRetryPolicy(
+ *          new ExponentialWithJitterRetryPolicy(
+ *              5, 1, 10, 1.5, 0.1,
+ *              [\Grpc\STATUS_CANCELLED,
+ *              \Grpc\STATUS_DEADLINE_EXCEEDED,
+ *              \Grpc\STATUS_PERMISSION_DENIED,
+ *              \Grpc\STATUS_RESOURCE_EXHAUSTED,
+ *              \Grpc\STATUS_ABORTED,
+ *              \Grpc\STATUS_OUT_OF_RANGE,
+ *              \Grpc\STATUS_UNAVAILABLE,
+ *              \Grpc\STATUS_DATA_LOSS,
+ *              \Grpc\STATUS_UNAUTHENTICATED,
+ *          ]));
+ *
+ *    The arguments in ExponentialWithJitterRetryPolicy() are as follows:
+ *      ExponentialWithJitterRetryPolicy(
+ *          defaultMaxAttempts, - total no of retry attempts allowed
+ *          initialBackoff,     - total no of secs to wait before retrying for the first time
+ *          maxBackoff,         - maximum no of sec to wait before retry timeout
+ *          backoffMultiplier,  - multiplier value used to calculate the delay before next attempt
+ *          jitter,             - jitter value to add randomness to delay value
+ *          retryableStatusCodes- array of retryable status code. All the status codes apart from this will not be retried
+ *      )
+ *
+ *    User can set the retryable status codes separately as well using following api
+ *    $this->setRetryableStatusCodes([
+ *      \Grpc\STATUS_CANCELLED,
+ *      \Grpc\STATUS_DEADLINE_EXCEEDED,
+ *      \Grpc\STATUS_PERMISSION_DENIED,
+ *      \Grpc\STATUS_RESOURCE_EXHAUSTED,
+ *      \Grpc\STATUS_ABORTED,
+ *      \Grpc\STATUS_OUT_OF_RANGE,
+ *      \Grpc\STATUS_UNAVAILABLE,
+ *      \Grpc\STATUS_DATA_LOSS,
+ *      \Grpc\STATUS_UNAUTHENTICATED,
+ *      ]);
+ *
+ *    User can also provide the delay scheduler with following api:
+ *    $this->setDelayScheduler(new NonBlockingScheduler());
+ *    This api will define whether delay function is blocking or non blocking
+ *    NonBlockingScheduler will be more useful when fiber is implemented.
+ *    By Default scheduler is of type BlockingScheduler.
+ */
+interface RetryPolicyInterface
+{
+    public const DEFAULT_MAX_ATTEMPTS = 5;
+    public const DEFAULT_INITIAL_BACKOFF = 1;
+    public const DEFAULT_MAX_BACKOFF = 10;
+    public const DEFAULT_BACKOFF_MULTIPLIER = 1.5;
+    public const DEFAULT_JITTER = 0.1;
+
+    /**
+     * Returns whether the request should be retried.
+     *
+     * @param int $attempt - current number of retry attempt
+     * @param int $status - SpanExporterStatusInterface
+     * @param RetryableExportException|null $exception: exception caught while retrying
+     *
+     * @return bool
+     */
+    public function shouldRetry(
+        int $attempt,
+        int $status,
+        ?RetryableExportException $exception
+    ): ?bool;
+
+    /**
+     * Returns the time to wait in seconds.
```

Comment:
Should be milliseconds.

Reply:
Yes it is actually in millisecond. Its a typo in comment. Corrected